scripts/publishCoverage.sh

```diff
@@ -14,83 +14,102 @@
+OLD_COVER_HTML="$COVERAGE_DIR/cover.html"
+OLD_COVERAGE=$(get_coverage "$OLD_COVER_HTML")
+
+OLD_COVERAGE=$(printf "%.2f" "$OLD_COVERAGE")
+NEW_COVERAGE=$(printf "%.2f" "$NEW_COVERAGE")
 
-# Compute overall coverage percentage
-echo "Computing the coverages"
-OLD_COVERAGE=$(cat $TRAVIS_BUILD_DIR/gh-pages/coverage/$TRAVIS_BRANCH/cover.html  | grep "%)"  | sed 's/[][()><%]/ /g' | awk '{ print $4 }' | awk '{s+=$1}END{print s/NR}')
-cp $TRAVIS_BUILD_DIR/cover.html $TRAVIS_BUILD_DIR/gh-pages/coverage/$TRAVIS_BRANCH
-cp $TRAVIS_BUILD_DIR/cover.html $TRAVIS_BUILD_DIR/gh-pages/coverage/$TRAVIS_COMMIT
-NEW_COVERAGE=$(cat $TRAVIS_BUILD_DIR/gh-pages/coverage/$TRAVIS_BRANCH/cover.html  | grep "%)"  | sed 's/[][()><%]/ /g' | awk '{ print $4 }' | awk '{s+=$1}END{print s/NR}')
+echo "===== Coverage comparison ====="
+echo "Old Coverage: $OLD_COVERAGE%"
+echo "New Coverage: $NEW_COVERAGE%"
+
+# update coverage reports
+mkdir -p "$COVERAGE_DIR"
+mkdir -p "coverage/$GITHUB_SHA"
+cp "$NEW_COVERAGE_SOURCE" "$COVERAGE_DIR/cover.html"
+cp "$NEW_COVERAGE_SOURCE" "coverage/$GITHUB_SHA/cover.html"
 
 if (( $(echo "$NEW_COVERAGE > $GREEN_THRESHOLD" | bc -l) )); then
-	BADGE_COLOR="green"
+    BADGE_COLOR="green"
 elif (( $(echo "$NEW_COVERAGE > $YELLOW_THRESHOLD" | bc -l) )); then
-	BADGE_COLOR="yellow"
+    BADGE_COLOR="yellow"
 fi
 
-# Generate badge for coverage
-curl https://img.shields.io/badge/coverage-$NEW_COVERAGE-$BADGE_COLOR.svg > $TRAVIS_BUILD_DIR/gh-pages/coverage/$TRAVIS_BRANCH/badge.svg
-
-COMMIT_RANGE=(${TRAVIS_COMMIT_RANGE//.../ })
+curl -s "https://img.shields.io/badge/coverage-${NEW_COVERAGE}%25-${BADGE_COLOR}.svg" \
+  > "$COVERAGE_DIR/badge.svg"
 
-# Generate result message for log and PR
+# coverage result message
 if (( $(echo "$OLD_COVERAGE > $NEW_COVERAGE" | bc -l) )); then
-	RESULT_MESSAGE=":red_circle: Coverage decreased from [$OLD_COVERAGE%] to [$NEW_COVERAGE%]"
+	RESULT_MESSAGE=":red_circle: Coverage decreased from **$OLD_COVERAGE%** → **$NEW_COVERAGE%**"
 elif (( $(echo "$OLD_COVERAGE == $NEW_COVERAGE" | bc -l) )); then
-	RESULT_MESSAGE=":thumbsup: Coverage remained same at [$NEW_COVERAGE%]"
+    RESULT_MESSAGE=":thumbsup: Coverage remained the same at **$NEW_COVERAGE%**"
 else
-	RESULT_MESSAGE=":thumbsup: Coverage increased from [$OLD_COVERAGE%] to [$NEW_COVERAGE%]"
+	RESULT_MESSAGE=":thumbsup: Coverage increased from **$OLD_COVERAGE%** → **$NEW_COVERAGE%**"
 fi
 
-# Update gh-pages branch or PR
-echo "Updating gh-pages"
-if [ "$TRAVIS_PULL_REQUEST" == "false" ]; then
-	git status
-	git add .
-	git commit -m "Coverage result for commit $TRAVIS_COMMIT from build $TRAVIS_BUILD_NUMBER"
-	git push origin
-else
-        # Updates PR with coverage
-   		curl -X POST -H "Authorization: Token $GHE_TOKEN" "https://api.github.com/repos/$TRAVIS_REPO_SLUG/issues/$TRAVIS_PULL_REQUEST/comments" -H 'Content-Type: application/json' --data '{"body": "'"$RESULT_MESSAGE"'"}'
-
+# update gh-pages or PR
+if [[ "$GITHUB_EVENT_NAME" == "push" ]]; then
```

Comment:
Note that pushes can also happen on pull requests if you create branches within the main repository.  You will get a push event for the push into the branch in the main repository, then a separate pull_request event if that branch is part of a pull request. If you creates branches on a fork, then this won't happen (or more accurately, the push event happens on the fork side, and the pull_request event happens on the main repo side.